### DISCUSSION TITLE
Do you need a MongoDB package or is the mongoose package sufficient?

### DISCUSSION THREAD

**Discussion by alinarub** [neuefische/questions#791] [Web] [answered]
Hello again,

We are in the process of implementing a database connection to MongoDB. We already created a cluster. Now Atlas is asking us to install a `mongodb` package. 

<img width="755" alt="Screenshot 2023-12-05 at 12 28 37" src="https://github.com/neuefische/questions/assets/137082094/a6c1c778-d4c4-4e97-8a5b-d4c8621e99e6">

Do we need this package or is the `mongoose` package enough? Do the`mongoose` and `mongodb` packages rule each other out or are both necessary for different scenarios? Once the database is running and we have the connection url we are good to go, but until then we need some support.

Thank you ☺️

**Comment by NikRoe** [comment]
Hi there,

you won't need the `mongodb` package. Going only with `mongoose` will suffice!

**Comment by gregorsart** [comment]
Hi there, 

we followed the instructions to connect MongoDB.
Unfortunately we get the following error when we try to read the dummy data in VSC we created in Atlas beforehand:

```Error [OverwriteModelError]: Cannot overwrite `Statistics` model once compiled.```

We appreciate every help.

**Reply by gregorsart** [reply]
Nevermind, we found the error for that: We spelled something wrong in:

```const Statistics =
  mongoose.models.Statistics || mongoose.model("Statistics", statisticsSchema);```

**Comment by gregorsart** [comment]
Hi , we progressed a little bit, but we get a response of: 
```Status Code:
304 Not Modified``
when we query the MongoDB database.

The data object we receive is an empty array, but is should be the number `26` we defined as dummy data in Atlas GUI.

**Comment by gregorsart** [comment, marked as answer]
With the help of Niklas we found the issue. We did not consider that the dummy data we initially created in Atlas GUI wasn't used at all by our `Schema` used in VSC with mongoose. Instead, Mongoose uses the VSC defined `Schema` and "pushes" it to Mongo DB as a new collection.
